### ISSUE TITLE
After upgrading I don't see grouped messages be Message Type.

### ISSUE THREAD

**Issue by pablocastilla** [Particular/ServicePulse#433]
Hi!

After upgrading I don't see failed message groups when I group by Message Type. By exception it works normally.

Is there anything I have to wait or execute?

***************

*Technical information if reporting bug:*

ServicePulse v1.8.4 • ServiceControl v1.38.0

Upgraded from Pulse 1.6.7 - SC 1.23

**Comment by WilliamBZA** [comment]
Hey again @pablocastilla 

The Message Type grouping is only applied to new messages coming in by default. If you want to apply it to existing messages, you can trigger the reclassification process by making a POST (no parameters are necessary) to `http://YOURSERVICECONTROLURLHERE/recoverability/groups/reclassify`

This should kick off the reclassification of the existing groups - which could take quite a while depending on the number of errors in your database.

**Comment by WilliamBZA** [comment]
Although, if you have a lot of messages in your database it might be worth waiting for our next release (hopefully today or tomorrow?) as we're introducing another classifier (https://github.com/Particular/ServicePulse/issues/427) in the next release

**Comment by pablocastilla** [comment]
William, thanks so much!

I think I can not wait that longer.

I have launched with PS:
Invoke-WebRequest -Uri http://localhost:33333/api/recoverability/groups/reclassify -Method POST 

And I don't see anything happening.

Is this normal? I have 88k error messages.

**Comment by WilliamBZA** [comment]
The process takes quite a while, once completed there should be an entry in your logfile reading:

`Reclassification of failures ended`

And then RavenDB will need some time to get all the indexes up to date. Is there high disk and cpu activity on your ServiceControl instance?

**Comment by pablocastilla** [comment]
No, I don't see anything. But I will give it a little time.

**Comment by WilliamBZA** [comment]
Hmm, and does the logfile show that the reclassification has completed?

**Comment by pablocastilla** [comment]
No, the last entry in the log file is much older.

I see a 202 in the post call and errors if I change stuff. 

What can I do?

**Comment by WilliamBZA** [comment]
That's strange. The 202 indicates Service Control has definitely received the request, but the lack of log file activity is worrying. Are you able to access the MSMQ queues on the ServiceControl machine? There should be a `particular.Servicecontrol` queue which receives a message after the 202. Are you able to see one there?

**Comment by pablocastilla** [comment]
I see the messages and how SC gets them.

I have restarted and retry everything without luck.

In the raven error I see the mapreduce errors and in the log I have this:

2017-03-30 16:34:27.0696|9|Error|ServiceBus.Management.Infrastructure.Nancy.NServiceBusContainerBootstrapper|Http call failed

System.InvalidOperationException: You cannot open a session or access the database commands before initializing the document store. Did you forget calling Initialize()?

   at Raven.Client.Document.DocumentStore.get_DatabaseCommands() in c:\Builds\RavenDB-Stable-2.5\Raven.Client.Lightweight\Document\DocumentStore.cs:line 115

   at Raven.Client.Document.DocumentStore.OpenSession(OpenSessionOptions options) in c:\Builds\RavenDB-Stable-2.5\Raven.Client.Lightweight\Document\DocumentStore.cs:line 358

   at ServiceControl.MessageFailures.Api.GetAllErrors.<.ctor>b__0_0(Object _) in C:\BuildAgent\work\7189a56f9f44affc\src\ServiceControl\MessageFailures\Api\GetAllErrors.cs:line 18

   at CallSite.Target(Closure , CallSite , Func`2 , Object )

   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)

   at Nancy.Routing.Route.<>c__DisplayClass4.<Wrap>b__3(Object parameters, CancellationToken context)

2017-03-30 16:34:27.0696|7|Error|ServiceBus.Management.Infrastructure.Nancy.NServiceBusContainerBootstrapper|Http call failed

System.InvalidOperationException: You cannot open a session or access the database commands before initializing the document store. Did you forget calling Initialize()?

   at Raven.Client.Document.DocumentStore.get_DatabaseCommands() in c:\Builds\RavenDB-Stable-2.5\Raven.Client.Lightweight\Document\DocumentStore.cs:line 115

   at Raven.Client.Document.DocumentStore.OpenSession(OpenSessionOptions options) in c:\Builds\RavenDB-Stable-2.5\Raven.Client.Lightweight\Document\DocumentStore.cs:line 358

   at ServiceControl.MessageFailures.Api.GetAllErrors.<.ctor>b__0_0(Object _) in C:\BuildAgent\work\7189a56f9f44affc\src\ServiceControl\MessageFailures\Api\GetAllErrors.cs:line 18

   at CallSite.Target(Closure , CallSite , Func`2 , Object )

   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)

   at Nancy.Routing.Route.<>c__DisplayClass4.<Wrap>b__3(Object parameters, CancellationToken context)

2017-03-30 16:34:27.0696|8|Error|ServiceBus.Management.Infrastructure.Nancy.NServiceBusContainerBootstrapper|Http call failed

System.InvalidOperationException: You cannot open a session or access the database commands before initializing the document store. Did you forget calling Initialize()?

   at Raven.Client.Document.DocumentStore.get_DatabaseCommands() in c:\Builds\RavenDB-Stable-2.5\Raven.Client.Lightweight\Document\DocumentStore.cs:line 115

   at Raven.Client.Document.DocumentStore.OpenSession(OpenSessionOptions options) in c:\Builds\RavenDB-Stable-2.5\Raven.Client.Lightweight\Document\DocumentStore.cs:line 358

   at ServiceControl.Recoverability.FailureGroupsApi.GetRetryHistory() in C:\BuildAgent\work\7189a56f9f44affc\src\ServiceControl\Recoverability\Grouping\Api\FailureGroupsApi.cs:line 90

   at CallSite.Target(Closure , CallSite , Func`2 , Object )

   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)

   at Nancy.Routing.Route.<>c__DisplayClass4.<Wrap>b__3(Object parameters, CancellationToken context)

2017-03-30 16:34:27.0696|3|Error|ServiceBus.Management.Infrastructure.Nancy.NServiceBusContainerBootstrapper|Http call failed

System.InvalidOperationException: You cannot open a session or access the database commands before initializing the document store. Did you forget calling Initialize()?

   at Raven.Client.Document.DocumentStore.get_DatabaseCommands() in c:\Builds\RavenDB-Stable-2.5\Raven.Client.Lightweight\Document\DocumentStore.cs:line 115

   at Raven.Client.Document.DocumentStore.OpenSession(OpenSessionOptions options) in c:\Builds\RavenDB-Stable-2.5\Raven.Client.Lightweight\Document\DocumentStore.cs:line 358

   at ServiceControl.Recoverability.FailureGroupsApi.GetAllGroups(String classifier) in C:\BuildAgent\work\7189a56f9f44affc\src\ServiceControl\Recoverability\Grouping\Api\FailureGroupsApi.cs:line 102

   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)

   at ServiceControl.Recoverability.FailureGroupsApi.<.ctor>b__12_2(Object parameters) in C:\BuildAgent\work\7189a56f9f44affc\src\ServiceControl\Recoverability\Grouping\Api\FailureGroupsApi.cs:line 32

   at CallSite.Target(Closure , CallSite , Func`2 , Object )

   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)

   at Nancy.Routing.Route.<>c__DisplayClass4.<Wrap>b__3(Object parameters, CancellationToken context)

2017-03-30 16:34:27.0696|11|Error|ServiceBus.Management.Infrastructure.Nancy.NServiceBusContainerBootstrapper|Http call failed

System.InvalidOperationException: You cannot open a session or access the database commands before initializing the document store. Did you forget calling Initialize()?

   at Raven.Client.Document.DocumentStore.get_DatabaseCommands() in c:\Builds\RavenDB-Stable-2.5\Raven.Client.Lightweight\Document\DocumentStore.cs:line 115

   at Raven.Client.Document.DocumentStore.OpenSession(OpenSessionOptions options) in c:\Builds\RavenDB-Stable-2.5\Raven.Client.Lightweight\Document\DocumentStore.cs:line 358

   at ServiceControl.MessageFailures.Api.GetAllErrors.<.ctor>b__0_0(Object _) in C:\BuildAgent\work\7189a56f9f44affc\src\ServiceControl\MessageFailures\Api\GetAllErrors.cs:line 18

   at CallSite.Target(Closure , CallSite , Func`2 , Object )

   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)

   at Nancy.Routing.Route.<>c__DisplayClass4.<Wrap>b__3(Object parameters, CancellationToken context)

2017-03-30 16:34:31.5058|6|Warn|NServiceBus.MessageInterfaces.MessageMapper.Reflection.MessageMapper|Interface ICustomAttributeProvider contains methods and can there for not be mapped. Be aware that non mapped interface can't be used to send messages.

2017-03-30 16:34:31.6933|9|Warn|ServiceControl.Operations.DetectFailedMessageImportsFeature|One or more error messages have previously failed to import properly into ServiceControl and have been stored in ServiceControl database.

Due to a defect however, ServiceControl would not be able to automatically reimport them. Please run ServiceControl in the maintenance mode and use embedded RavenStudio available by default at http://localhost:33333/storage to examine the payloads of failed messages to ensure no information has been lost.

Delete the failed import documents afterwards so that you don't see this warning message again.

**Comment by WilliamBZA** [comment]
We may have to dig into raven then and see why the index errors are occurring.

Can you switch Service Control into [maintenance mode](https://docs.particular.net/servicecontrol/use-ravendb-studio), open raven studio, go to the status page and see how many stale indexes there are?

Alternatively, is it possible to share the database with us so we can take a look?

**Comment by pablocastilla** [comment]
It says 0 stale indexes.

Sharing the ddbb would be more difficult as I would need to request permission for that.

![raven](https://cloud.githubusercontent.com/assets/8155105/24538932/5c24f666-15ec-11e7-8bb5-97b3eb55530f.PNG)

**Comment by pablocastilla** [comment]
I have switched again to Maintenance Mode and I see two stales:
Auto/EventLogItems/ByRaisedAtSortByRaisedAt
Auto/ExternalIntegrationDispatchRequest

Also in the log I see the following warnings:
Could not open index sagadetailsindex, recovery operation failed. --> map reduce exception
Could not open index sagalistindex, recovery operation failed. --> map reduce exception

**Comment by pablocastilla** [comment]
I tried a compaction but when I tried to restore it it didn't work.

I will open a support ticket on Monday. This database has been upgraded a few times, maybe something is missing on it.

**Comment by WilliamBZA** [comment]
@pablocastilla I did find a bug with the reclassification code (which is now fixed in 1.39.0), but I don't think it's the cause of your problems. We'd still see the log messages even with the bug.

**Comment by pablocastilla** [comment]
Thanks for pointing me to 1.39.

I will do the following:

1.- Try the 1.39 version.
2.- If it doesn't work I will ask the client for permission to share the db with you. I would open a support ticket and tell you that it is open here.
3.- Wait your answer :D

what do you think?

**Comment by WilliamBZA** [comment]
I think that's a good plan. There's definitely some oddity going on with the database, being able to inspect it will go a long way towards identifying the problem.

**Comment by pablocastilla** [comment]
ummmmmmmmm, now I see the messages but when I want to retry them it fails with the pasted exception. I will ask for permission to give you the database.

2017-04-04 09:15:32.2034|24|Fatal|NServiceBus|Failed to import too many times

System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.

   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)

   at ServiceControl.Operations.FailedMessageFactory.ParseFailureDetails(IReadOnlyDictionary`2 headers) in C:\BuildAgent\work\7189a56f9f44affc\src\ServiceControl\Operations\FailedMessageFactory.cs:line 39

   at ServiceControl.Operations.ErrorQueueImport.InnerHandle(TransportMessage message) in C:\BuildAgent\work\7189a56f9f44affc\src\ServiceControl\Operations\ErrorQueueImport.cs:line 138

   at ServiceControl.Operations.ErrorQueueImport.Handle(TransportMessage message) in C:\BuildAgent\work\7189a56f9f44affc\src\ServiceControl\Operations\ErrorQueueImport.cs:line 81

   at NServiceBus.Satellites.SatelliteLauncher.HandleMessageReceived(Object sender, TransportMessageReceivedEventArgs e, ISatellite satellite) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Satellites\SatelliteLauncher.cs:line 89

   at NServiceBus.Unicast.Transport.TransportReceiver.ProcessMessage(TransportMessage message) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Unicast\Transport\TransportReceiver.cs:line 346

   at NServiceBus.Unicast.Transport.TransportReceiver.TryProcess(TransportMessage message) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Unicast\Transport\TransportReceiver.cs:line 230

   at NServiceBus.Transports.Msmq.MsmqDequeueStrategy.Action(TransactionSettings transactionSettings, TransactionOptions transactionOptions, MsmqUnitOfWork unitOfWork, MessageQueue receiveQueue, MessageQueue errorQueue, CircuitBreaker circuitBreaker, CriticalError criticalError, AutoResetEvent peekResetEvent, TimeSpan receiveTimeout, SemaphoreSlim throttlingSemaphore, Func`2 tryProcessMessage, Action`2 endProcessMessage) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Transports\Msmq\MsmqDequeueStrategy.cs:line 229

2017-04-04 09:15:32.2304|124|Fatal|NServiceBus|Failed to import too many times

System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.

   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)

   at ServiceControl.Operations.FailedMessageFactory.ParseFailureDetails(IReadOnlyDictionary`2 headers) in C:\BuildAgent\work\7189a56f9f44affc\src\ServiceControl\Operations\FailedMessageFactory.cs:line 39

   at ServiceControl.Operations.ErrorQueueImport.InnerHandle(TransportMessage message) in C:\BuildAgent\work\7189a56f9f44affc\src\ServiceControl\Operations\ErrorQueueImport.cs:line 138

   at ServiceControl.Operations.ErrorQueueImport.Handle(TransportMessage message) in C:\BuildAgent\work\7189a56f9f44affc\src\ServiceControl\Operations\ErrorQueueImport.cs:line 81

   at NServiceBus.Satellites.SatelliteLauncher.HandleMessageReceived(Object sender, TransportMessageReceivedEventArgs e, ISatellite satellite) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Satellites\SatelliteLauncher.cs:line 89

   at NServiceBus.Unicast.Transport.TransportReceiver.ProcessMessage(TransportMessage message) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Unicast\Transport\TransportReceiver.cs:line 346

   at NServiceBus.Unicast.Transport.TransportReceiver.TryProcess(TransportMessage message) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Unicast\Transport\TransportReceiver.cs:line 230

   at NServiceBus.Transports.Msmq.MsmqDequeueStrategy.Action(TransactionSettings transactionSettings, TransactionOptions transactionOptions, MsmqUnitOfWork unitOfWork, MessageQueue receiveQueue, MessageQueue errorQueue, CircuitBreaker circuitBreaker, CriticalError criticalError, AutoResetEvent peekResetEvent, TimeSpan receiveTimeout, SemaphoreSlim throttlingSemaphore, Func`2 tryProcessMessage, Action`2 endProcessMessage) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Transports\Msmq\MsmqDequeueStrategy.cs:line 229

2017-04-04 09:15:32.2464|115|Fatal|NServiceBus|Failed to import too many times

System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.

   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)

   at ServiceControl.Operations.FailedMessageFactory.ParseFailureDetails(IReadOnlyDictionary`2 headers) in C:\BuildAgent\work\7189a56f9f44affc\src\ServiceControl\Operations\FailedMessageFactory.cs:line 39

   at ServiceControl.Operations.ErrorQueueImport.InnerHandle(TransportMessage message) in C:\BuildAgent\work\7189a56f9f44affc\src\ServiceControl\Operations\ErrorQueueImport.cs:line 138

   at ServiceControl.Operations.ErrorQueueImport.Handle(TransportMessage message) in C:\BuildAgent\work\7189a56f9f44affc\src\ServiceControl\Operations\ErrorQueueImport.cs:line 81

   at NServiceBus.Satellites.SatelliteLauncher.HandleMessageReceived(Object sender, TransportMessageReceivedEventArgs e, ISatellite satellite) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Satellites\SatelliteLauncher.cs:line 89

   at NServiceBus.Unicast.Transport.TransportReceiver.ProcessMessage(TransportMessage message) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Unicast\Transport\TransportReceiver.cs:line 346

   at NServiceBus.Unicast.Transport.TransportReceiver.TryProcess(TransportMessage message) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Unicast\Transport\TransportReceiver.cs:line 230

   at NServiceBus.Transports.Msmq.MsmqDequeueStrategy.Action(TransactionSettings transactionSettings, TransactionOptions transactionOptions, MsmqUnitOfWork unitOfWork, MessageQueue receiveQueue, MessageQueue errorQueue, CircuitBreaker circuitBreaker, CriticalError criticalError, AutoResetEvent peekResetEvent, TimeSpan receiveTimeout, SemaphoreSlim throttlingSemaphore, Func`2 tryProcessMessage, Action`2 endProcessMessage) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Transports\Msmq\MsmqDequeueStrategy.cs:line 229

2017-04-04 09:15:32.2634|69|Fatal|NServiceBus|Failed to import too many times

System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.

   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)

   at ServiceControl.Operations.FailedMessageFactory.ParseFailureDetails(IReadOnlyDictionary`2 headers) in C:\BuildAgent\work\7189a56f9f44affc\src\ServiceControl\Operations\FailedMessageFactory.cs:line 39

   at ServiceControl.Operations.ErrorQueueImport.InnerHandle(TransportMessage message) in C:\BuildAgent\work\7189a56f9f44affc\src\ServiceControl\Operations\ErrorQueueImport.cs:line 138

   at ServiceControl.Operations.ErrorQueueImport.Handle(TransportMessage message) in C:\BuildAgent\work\7189a56f9f44affc\src\ServiceControl\Operations\ErrorQueueImport.cs:line 81

   at NServiceBus.Satellites.SatelliteLauncher.HandleMessageReceived(Object sender, TransportMessageReceivedEventArgs e, ISatellite satellite) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Satellites\SatelliteLauncher.cs:line 89

   at NServiceBus.Unicast.Transport.TransportReceiver.ProcessMessage(TransportMessage message) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Unicast\Transport\TransportReceiver.cs:line 346

   at NServiceBus.Unicast.Transport.TransportReceiver.TryProcess(TransportMessage message) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Unicast\Transport\TransportReceiver.cs:line 230

   at NServiceBus.Transports.Msmq.MsmqDequeueStrategy.Action(TransactionSettings transactionSettings, TransactionOptions transactionOptions, MsmqUnitOfWork unitOfWork, MessageQueue receiveQueue, MessageQueue errorQueue, CircuitBreaker circuitBreaker, CriticalError criticalError, AutoResetEvent peekResetEvent, TimeSpan receiveTimeout, SemaphoreSlim throttlingSemaphore, Func`2 tryProcessMessage, Action`2 endProcessMessage) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Transports\Msmq\MsmqDequeueStrategy.cs:line 229

2017-04-04 09:15:32.2775|19|Fatal|NServiceBus|Failed to import too many times

System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.

   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)

   at ServiceControl.Operations.FailedMessageFactory.ParseFailureDetails(IReadOnlyDictionary`2 headers) in C:\BuildAgent\work\7189a56f9f44affc\src\ServiceControl\Operations\FailedMessageFactory.cs:line 39

   at ServiceControl.Operations.ErrorQueueImport.InnerHandle(TransportMessage message) in C:\BuildAgent\work\7189a56f9f44affc\src\ServiceControl\Operations\ErrorQueueImport.cs:line 138

   at ServiceControl.Operations.ErrorQueueImport.Handle(TransportMessage message) in C:\BuildAgent\work\7189a56f9f44affc\src\ServiceControl\Operations\ErrorQueueImport.cs:line 81

   at NServiceBus.Satellites.SatelliteLauncher.HandleMessageReceived(Object sender, TransportMessageReceivedEventArgs e, ISatellite satellite) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Satellites\SatelliteLauncher.cs:line 89

   at NServiceBus.Unicast.Transport.TransportReceiver.ProcessMessage(TransportMessage message) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Unicast\Transport\TransportReceiver.cs:line 346

   at NServiceBus.Unicast.Transport.TransportReceiver.TryProcess(TransportMessage message) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Unicast\Transport\TransportReceiver.cs:line 230

   at NServiceBus.Transports.Msmq.MsmqDequeueStrategy.Action(TransactionSettings transactionSettings, TransactionOptions transactionOptions, MsmqUnitOfWork unitOfWork, MessageQueue receiveQueue, MessageQueue errorQueue, CircuitBreaker circuitBreaker, CriticalError criticalError, AutoResetEvent peekResetEvent, TimeSpan receiveTimeout, SemaphoreSlim throttlingSemaphore, Func`2 tryProcessMessage, Action`2 endProcessMessage) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Transports\Msmq\MsmqDequeueStrategy.cs:line 229

2017-04-04 09:15:32.2775|3|Error|NServiceBus.Transports.Msmq.MsmqDequeueStrategy|Error processing message.

System.ObjectDisposedException: The semaphore has been disposed.

   at System.Threading.SemaphoreSlim.Release(Int32 releaseCount)

   at NServiceBus.Transports.Msmq.MsmqDequeueStrategy.Action(TransactionSettings transactionSettings, TransactionOptions transactionOptions, MsmqUnitOfWork unitOfWork, MessageQueue receiveQueue, MessageQueue errorQueue, CircuitBreaker circuitBreaker, CriticalError criticalError, AutoResetEvent peekResetEvent, TimeSpan receiveTimeout, SemaphoreSlim throttlingSemaphore, Func`2 tryProcessMessage, Action`2 endProcessMessage)

   at NServiceBus.Transports.Msmq.MsmqDequeueStrategy.<OnPeekCompleted>b__12_1()

   at System.Threading.Tasks.Task.Execute()

2017-04-04 09:15:32.2775|49|Error|NServiceBus.Transports.Msmq.MsmqDequeueStrategy|Error processing message.

System.ObjectDisposedException: The semaphore has been disposed.

   at System.Threading.SemaphoreSlim.Release(Int32 releaseCount)

   at NServiceBus.Transports.Msmq.MsmqDequeueStrategy.Action(TransactionSettings transactionSettings, TransactionOptions transactionOptions, MsmqUnitOfWork unitOfWork, MessageQueue receiveQueue, MessageQueue errorQueue, CircuitBreaker circuitBreaker, CriticalError criticalError, AutoResetEvent peekResetEvent, TimeSpan receiveTimeout, SemaphoreSlim throttlingSemaphore, Func`2 tryProcessMessage, Action`2 endProcessMessage)

   at NServiceBus.Transports.Msmq.MsmqDequeueStrategy.<OnPeekCompleted>b__12_1()

   at System.Threading.Tasks.Task.Execute()

2017-04-04 09:15:32.2775|26|Error|NServiceBus.Unicast.Transport.TransportReceiver|Failed raising 'finished message processing' event.

System.ObjectDisposedException: Instances cannot be resolved and nested lifetimes cannot be created from this LifetimeScope as it has already been disposed.

   at Autofac.Core.Lifetime.LifetimeScope.ResolveComponent(IComponentRegistration registration, IEnumerable`1 parameters)

   at Autofac.ResolutionExtensions.ResolveService(IComponentContext context, Service service, IEnumerable`1 parameters)

   at NServiceBus.ObjectBuilder.Common.CommonObjectBuilder.Build[T]()

   at NServiceBus.Unicast.UnicastBus.TransportFinishedMessageProcessing(Object sender, FinishedMessageProcessingEventArgs e)

   at NServiceBus.Unicast.Transport.TransportReceiver.ProcessMessage(TransportMessage message)

2017-04-04 09:15:32.2945|19|Error|NServiceBus.Transports.Msmq.MsmqDequeueStrategy|Error processing message.

System.ObjectDisposedException: The semaphore has been disposed.

   at System.Threading.SemaphoreSlim.Release(Int32 releaseCount)

   at NServiceBus.Transports.Msmq.MsmqDequeueStrategy.Action(TransactionSettings transactionSettings, TransactionOptions transactionOptions, MsmqUnitOfWork unitOfWork, MessageQueue receiveQueue, MessageQueue errorQueue, CircuitBreaker circuitBreaker, CriticalError criticalError, AutoResetEvent peekResetEvent, TimeSpan receiveTimeout, SemaphoreSlim throttlingSemaphore, Func`2 tryProcessMessage, Action`2 endProcessMessage)

   at NServiceBus.Transports.Msmq.MsmqDequeueStrategy.<OnPeekCompleted>b__12_1()

   at System.Threading.Tasks.Task.Execute()

2017-04-04 09:15:32.2945|55|Error|NServiceBus.Transports.Msmq.MsmqDequeueStrategy|Error processing message.

System.ObjectDisposedException: The semaphore has been disposed.

   at System.Threading.SemaphoreSlim.Release(Int32 releaseCount)

   at NServiceBus.Transports.Msmq.MsmqDequeueStrategy.Action(TransactionSettings transactionSettings, TransactionOptions transactionOptions, MsmqUnitOfWork unitOfWork, MessageQueue receiveQueue, MessageQueue errorQueue, CircuitBreaker circuitBreaker, CriticalError criticalError, AutoResetEvent peekResetEvent, TimeSpan receiveTimeout, SemaphoreSlim throttlingSemaphore, Func`2 tryProcessMessage, Action`2 endProcessMessage)

   at NServiceBus.Transports.Msmq.MsmqDequeueStrategy.<OnPeekCompleted>b__12_1()

   at System.Threading.Tasks.Task.Execute()

2017-04-04 09:15:32.4585|122|Error|NServiceBus.Unicast.Transport.TransportReceiver|Failed raising 'finished message processing' event.

System.ObjectDisposedException: Instances cannot be resolved and nested lifetimes cannot be created from this LifetimeScope as it has already been disposed.

   at Autofac.Core.Lifetime.LifetimeScope.ResolveComponent(IComponentRegistration registration, IEnumerable`1 parameters)

   at Autofac.ResolutionExtensions.ResolveService(IComponentContext context, Service service, IEnumerable`1 parameters)

   at NServiceBus.ObjectBuilder.Common.CommonObjectBuilder.Build[T]()

   at NServiceBus.Unicast.UnicastBus.TransportFinishedMessageProcessing(Object sender, FinishedMessageProcessingEventArgs e)

   at NServiceBus.Unicast.Transport.TransportReceiver.ProcessMessage(TransportMessage message)

2017-04-04 09:15:32.4715|59|Error|NServiceBus.Unicast.Transport.TransportReceiver|Failed raising 'finished message processing' event.

System.ObjectDisposedException: Instances cannot be resolved and nested lifetimes cannot be created from this LifetimeScope as it has already been disposed.

   at Autofac.Core.Lifetime.LifetimeScope.ResolveComponent(IComponentRegistration registration, IEnumerable`1 parameters)

   at Autofac.ResolutionExtensions.ResolveService(IComponentContext context, Service service, IEnumerable`1 parameters)

   at NServiceBus.ObjectBuilder.Common.CommonObjectBuilder.Build[T]()

   at NServiceBus.Unicast.UnicastBus.TransportFinishedMessageProcessing(Object sender, FinishedMessageProcessingEventArgs e)

   at NServiceBus.Unicast.Transport.TransportReceiver.ProcessMessage(TransportMessage message)

2017-04-04 09:15:32.4715|29|Error|NServiceBus.Transports.Msmq.MsmqDequeueStrategy|Error processing message.

System.ObjectDisposedException: The semaphore has been disposed.

   at System.Threading.SemaphoreSlim.Release(Int32 releaseCount)

   at NServiceBus.Transports.Msmq.MsmqDequeueStrategy.Action(TransactionSettings transactionSettings, TransactionOptions transactionOptions, MsmqUnitOfWork unitOfWork, MessageQueue receiveQueue, MessageQueue errorQueue, CircuitBreaker circuitBreaker, CriticalError criticalError, AutoResetEvent peekResetEvent, TimeSpan receiveTimeout, SemaphoreSlim throttlingSemaphore, Func`2 tryProcessMessage, Action`2 endProcessMessage)

   at NServiceBus.Transports.Msmq.MsmqDequeueStrategy.<OnPeekCompleted>b__12_1()

   at System.Threading.Tasks.Task.Execute()

**Comment by pablocastilla** [comment]
Also now SC doesn't start :S

**Comment by WilliamBZA** [comment]
Those exceptions occur during message importing, something is going wrong while importing the messages that are in the error queue. Let's get a support ticket going so we can get on a call together.

**Comment by pablocastilla** [comment]
Created, but I don't get any email with the support ticket number.

**Comment by WilliamBZA** [comment]
@pablocastilla The original issue here has been handled, and the new issue is being dealt with as a support case. I'll reach out there once we have a reliable fix in place.
